# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/AccGroupInfo.h
# Commit: d46f0bd36d7e
# Full Hash: d46f0bd36d7edd1d9d040be6298e6e5d1ccc70b2
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 1
# Description:
#   Bug 1737944 - P9: Make AccGroupInfo use Accessible. r=Jamie
#   
#   Now that GroupPosition and GetGroupInfo are in Accessible, we can make
#   AccGroupInfo use the Accessible base class.
#   
# ==============================================================================

diff -r ac04b727a98b -r d46f0bd36d7e accessible/base/AccGroupInfo.h
--- a/accessible/base/AccGroupInfo.h	Tue Jan 04 21:01:37 2022 +0000
+++ b/accessible/base/AccGroupInfo.h	Tue Jan 04 21:01:38 2022 +0000
@@ -11,7 +11,7 @@
 namespace mozilla {
 namespace a11y {
 
-class LocalAccessible;
+class Accessible;
 
 /**
  * Calculate and store group information.
@@ -38,7 +38,7 @@
    * Return a direct or logical parent of the accessible that this group info is
    * created for.
    */
-  LocalAccessible* ConceptualParent() const { return mParent; }
+  Accessible* ConceptualParent() const { return mParent; }
 
   /**
    * Update group information.
@@ -48,27 +48,26 @@
   /**
    * Create group info.
    */
-  static AccGroupInfo* CreateGroupInfo(const LocalAccessible* aAccessible);
+  static AccGroupInfo* CreateGroupInfo(const Accessible* aAccessible);
 
   /**
    * Return a first item for the given container.
    */
-  static LocalAccessible* FirstItemOf(const LocalAccessible* aContainer);
+  static Accessible* FirstItemOf(const Accessible* aContainer);
 
   /**
    * Return total number of items in container, and if it is has nested
    * collections.
    */
-  static uint32_t TotalItemCount(LocalAccessible* aContainer,
-                                 bool* aIsHierarchical);
+  static uint32_t TotalItemCount(Accessible* aContainer, bool* aIsHierarchical);
 
   /**
    * Return next item of the same group to the given item.
    */
-  static LocalAccessible* NextItemTo(LocalAccessible* aItem);
+  static Accessible* NextItemTo(Accessible* aItem);
 
  protected:
-  AccGroupInfo(const LocalAccessible* aItem, a11y::role aRole);
+  AccGroupInfo(const Accessible* aItem, a11y::role aRole);
 
  private:
   AccGroupInfo(const AccGroupInfo&) = delete;
@@ -98,12 +97,12 @@
    * Return ARIA level value or the default one if ARIA is missed for the
    * given accessible.
    */
-  static int32_t GetARIAOrDefaultLevel(const LocalAccessible* aAccessible);
+  static int32_t GetARIAOrDefaultLevel(const Accessible* aAccessible);
 
   uint32_t mPosInSet;
   uint32_t mSetSize;
-  LocalAccessible* mParent;
-  const LocalAccessible* mItem;
+  Accessible* mParent;
+  const Accessible* mItem;
   a11y::role mRole;
 };
 